docs/modules/ROOT/pages/quarkus-roq-data.adoc

```diff
@@ -0,0 +1,110 @@
+= Quarkus Roq Data
+
+TIP: Quarkus Roq Data is a Quarkus extension that uses JSON/YAML file content to use in a type-safe way in your project.
+
+== Installation
+
+If you want to use this extension, you need to add the `io.quarkiverse.roq:quarkus-roq-data` extension first to your build file.
+
+For instance, with Maven, add the following dependency to your POM file:
+
+[source,xml,subs=attributes+]
+----
+<dependency>
+    <groupId>io.quarkiverse.roq</groupId>
+    <artifactId>quarkus-roq-data</artifactId>
+    <version>{project-version}</version>
+</dependency>
+----
+
+== Getting started
+
+Once you have the extension installed, you can add a JSON or YAML file to your project's `src/main/site/data` directory.
+
+NOTE: The data file can use `.json`, `.yml`, or `.yaml` extensions.
+
+[source,json]
+.data.json
+----
+{
+    "name": "Quarkus Site",
+    "description": "Quarkus Site generated by Roq"
+}
+----
+
+To use the data in your Java code, you need to create a Java class that matches the structure of the JSON or YAML file.
+
+[source,java]
+.Data.java
+----
+import io.quarkiverse.roq.data.runtime.annotations.DataMapping;
+
+@DataMapping("data")
+public record Data(String name, String description) {}
+----
+
+Inject the `Data` record into a resource class.
```

Comment:
My only comment is the Data pojo too generic. Should it be something real world like Car. Only because users might think it needs to be called Data or something like that?